mlir/lib/Transforms/lab4/bonyuk_peter/BonyukFusedMultiplyAddPass.cpp

```diff
@@ -0,0 +1,75 @@
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Tools/Plugins/PassPlugin.h"
+
+using namespace mlir;
+
+namespace {
+class BonyukFusedMultiplyAddPass
+    : public PassWrapper<BonyukFusedMultiplyAddPass, OperationPass<ModuleOp>> {
+public:
+  StringRef getArgument() const final { return "bonyuk_fused_multiply_add"; }
+  StringRef getDescription() const final {
+    return "This Pass combines the operations of addition and multiplication "
+           "into one";
+  }
+
+  void runOnOperation() override {
+    ModuleOp module = getOperation();
+    module.walk([&](Operation *operation) {
+      if (auto AddOperation = dyn_cast<LLVM::FAddOp>(operation)) {
```

Comment:
No need to walk through all operations, you can filter them by type in the walker itself